apps/api/src/app.module.ts

```diff
@@ -134,6 +133,10 @@ if (process.env.SENTRY_DSN) {
   });
 }
 
+if (process.env.SEGMENT_TOKEN) {
+  modules.push(AnalyticsModule);
+}
```

Comment:
@davidsoderberg won't it break all self-hosted applications? since they are trying to import not available module? (analytics service)

Reply:
cc. @SokratisVidros

Reply:
No we use analytics service everywhere or have we moved it to enterprise?

Reply:
We import it everywhere. It's just not reporting anything (returning the track) method immediately. I believe this change will break the API in the current state.

My suggestion is leave it as it was before the commit. And let's merge this PR already :)

Reply:
Are you talking about the AnalyticsService or the AnalyticsModule introduced in this PR? The service is still imported as ESM so I am not sure I understand how the conditional import breaks it.

Reply:
The Analytics Module exports a service called AnalyticsService, we inject this service in many usecases in the code. If the module is not present, all those injections will fail in boot time because the service is not available. (or at least that's what I think :P )

Reply:
nvm, i thought the the analyticsmodule exposes the analytics service. in practices the analytics service is living in SharedModule. Analytics module is just the new endpoint for proxying 😅